tests/e2e/utils/parser.go

```diff
@@ -0,0 +1,79 @@
+// (c) 2022 Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package utils
+
+import (
+	"fmt"
+	"io/ioutil"
+	"strings"
+
+	"github.com/fatih/color"
+	"gopkg.in/yaml.v2"
+)
+
+/*
+===Example File===
+
+endpoint: /ext/bc/2Z36RnQuk1hvsnFeGWzfZUfXNr7w1SjzmDQ78YxfTVNAkDq3nZ
+logsDir: /var/folders/mp/6jm81gc11dv3xtcwxmrd8mcr0000gn/T/runnerlogs2984620995
+pid: 55547
+uris:
+- http://localhost:61278
+- http://localhost:61280
+- http://localhost:61282
+- http://localhost:61284
+- http://localhost:61286
+*/
+
+type output struct {
+	Endpoint string   `yaml:"endpoint"`
+	Logs     string   `yaml:"logsDir"`
+	PID      int      `yaml:"pid"`
+	URIs     []string `yaml:"uris"`
+}
+
+const (
+	HARDHAT_CONFIG = "contract-examples/hardhat.config.ts"
+)
+
+func updateHardhat(rpc string) {
+	input, err := ioutil.ReadFile(HARDHAT_CONFIG)
```

Comment:
I think we can create a separate hardhat config instead of updating existing one. hardhat can take `--config` as a specifiable option.

Reply:
I rewrote this. Instead of overriding the hardhat config, I write the rpc url to a git ignored file and read it from that file if it exists